docs/user-manual/en/broker-balancers.md

```diff
@@ -0,0 +1,156 @@
+By default, the cache is enabled, to disable the cache the `cache-timeout` parameter must be `0`.
+
+## Defining broker balancers
+A broker balancer is defined by `broker-balancer` element, it includes the following items:
+* the `name` attribute defines the name of the broker balancer;
+* the `target-key` element defines what key to select a target broker, the supported values are: `SNI_HOST`, `SOURCE_IP`, `USER_NAME`.
+* the `local-filter` element defines a regular expression to filter the keys that have to return a local target;
+* the `cache-timeout` element is the time period for a target broker to remain in the cache;
+* the `pool` element defines the pool to group the target brokers, see [pools](#pools).
+* the `policy` element defines the policy used to select the target brokers, see [policies](#policies);
+
+Let's take a look at some broker balancer examples from broker.xml:
+```xml
+<broker-balancers>
+    <broker-balancer name="simple-balancer">
+        <policy name="FIRST_ELEMENT"/>
+        <pool>
+            <static-connectors>
+                <connector-ref>connector1</connector-ref>
+            </static-connectors>
+        </pool>
+    </broker-balancer>
+    <broker-balancer name="consistent-hash-balancer">
+        <target-key>USER_NAME</target-key>
+        <local-filter>admin</local-filter>
+        <policy name="CONSISTENT_HASH"/>
+        <pool>
+            <local-target-enabled>true</local-target-enabled>
+            <discovery-group-ref discovery-group-name="dg1"/>
+        </pool>
+    <policy name="CONSISTENT_HASH"/>
+    </broker-balancer>
+    <broker-balancer name="least-connections-balancer">
+        <policy name="LEAST_CONNECTIONS"/>
+        <pool>
+            <discovery-group-ref discovery-group-name="dg2"/>
+        </pool>
+    </broker-balancer>
+</broker-balancers>
+```
+
+## Broker Balancer Workflow
+The broker balancer workflow include the following steps:
+* Retrieve the target key from the incoming connection;
+* Return the local target broker if the target key matches the local filter;
+* Return the cached target broker if it is ready;
+* Get ready target brokers from the pool;
+* Select one target broker using the policy;
+* Add the selected broker in the cache;
+* Return the selected broker.
+
+Let's take a look at flowchart of the broker balancer workflow:
+![Broker Balancer Workflow](images/broker_balancer_workflow.png)
+
+
+## Redirection
+Apache ActiveMQ Artemis provides a native redirection for supported clients and a new management API for other clients.
+The native redirection can be enabled per acceptor and is supported only for CORE and AMQP clients.
+The acceptor with the `redirect-to` url parameter will redirect the incoming connections.
+The `redirect-to` url parameter specifies the name of the broker balancer to use,
+ie the following acceptor will redirect the incoming CORE client connections using the broker balancer with the name `bb1`:
```

Comment:
maybe it makes sense to reuse the name from above. in place of bb1 use 'consistent-hash-balancer' just to tie the url parameter to the name element of the xml. If I have that correct!